providers/dns/transip/transip_test.go

```diff
@@ -135,6 +203,124 @@ func TestNewDNSProviderConfig(t *testing.T) {
 	}
 }
 
+func TestDNSProvider_concurrentGetInfo(t *testing.T) {
+	client := &fakeClient{
+		getInfoLatency:       50 * time.Millisecond,
+		setDNSEntriesLatency: 500 * time.Millisecond,
+	}
+
+	p := &DNSProvider{
+		config: NewDefaultConfig(),
+		client: client,
+	}
+
+	var wg sync.WaitGroup
+	wg.Add(2)
+
+	go func() {
+		defer wg.Done()
+
+		domain1 := "bar.lego.wtf"
+
+		time.Sleep(500 * time.Millisecond)
+		err := p.Present(domain1, "", "")
+		require.NoError(t, err)
+
+		time.Sleep(100 * time.Millisecond)
+		var found bool
+		for _, entry := range client.dnsEntries {
+			if strings.HasSuffix(entry.Name, ".bar") {
+				found = true
+			}
+		}
+		assert.True(t, found, "record bar not found: %v", client.dnsEntries)
+
+		time.Sleep(100 * time.Millisecond)
+		err = p.CleanUp(domain1, "", "")
+		require.NoError(t, err)
+	}()
+
+	go func() {
+		defer wg.Done()
+
+		domain2 := "foo.lego.wtf"
```

Comment:
Why not move this test code into a function to reduce the duplication of logic here?